api/admin/controller.py

```diff
@@ -321,40 +321,51 @@ def edit(self, data_source, identifier_type, identifier):
             work.calculate_presentation(policy=policy)
         return Response("", 200)
 
-    def suppress(self, data_source, identifier_type, identifier):
+    def suppress(self, identifier_type, identifier):
         """Suppress the license pool associated with a book."""
-        
         # Turn source + identifier into a LicensePool
-        pool = self.load_licensepool(data_source, identifier_type, identifier)
-        if isinstance(pool, ProblemDetail):
+        pools = self.load_licensepools(self.library, identifier_type, identifier)
+        if isinstance(pools, ProblemDetail):
             # Something went wrong.
-            return pool
-    
-        pool.suppressed = True
+            return pools
+
+        # Assume that the Work is being suppressed from the catalog, and
+        # not just the LicensePool.
+        # TODO: Suppress individual LicensePools when it's not that deep.
+        for pool in pools:
```

Comment:
I've filed https://github.com/NYPL-Simplified/server_core/issues/497 to address this.